src/routes/user/register/1.svelte

```diff
@@ -0,0 +1,384 @@
+	import { cn } from '$functions/classnames';
+	import { autofocus } from '$functions/forms/autofocus';
+	import Arrow from '$icons/shapes/arrow.svelte';
+	import Info from '$icons/shapes/info.svelte';
+	import Tick from '$icons/shapes/tick.svelte';
+	import ArrowUpRight from '$icons/shapes/arrow_up_right.svelte';
+	import CoreText from '$icons/text/core.svelte';
+	import type { FormFieldState, PageProps } from './types';
+
+	let { on_submit, pages_state }: PageProps = $props();
+
+	// let pages_state = new Array({});
+
+	let combined_state: { [key: string]: string };
+
+	$effect.pre(() => {
+		$inspect(pages_state);
+		combined_state = Object.assign({}, ...Object.entries(pages_state));
+	});
+
+	let confirm_password_element: HTMLInputElement | null = null;
+
+	const register_languages = async () => {
+		const zxcvbnCommonPackage = await import('@zxcvbn-ts/language-common');
+		const zxcvbnEnPackage = await import('@zxcvbn-ts/language-en');
+		const zxcvbnCsPackage = await import('@zxcvbn-ts/language-cs');
+		const zxcvbnDePackage = await import('@zxcvbn-ts/language-de');
+		const zxcvbnEsEsPackage = await import('@zxcvbn-ts/language-es-es');
+		const zxcvbnFiPackage = await import('@zxcvbn-ts/language-fi');
+		const zxcvbnFrPackage = await import('@zxcvbn-ts/language-fr');
+		const zxcvbnIdPackage = await import('@zxcvbn-ts/language-id');
+		const zxcvbnItPackage = await import('@zxcvbn-ts/language-it');
+		const zxcvbnJaPackage = await import('@zxcvbn-ts/language-ja');
+		const zxcvbnNlBePackage = await import('@zxcvbn-ts/language-nl-be');
+		const zxcvbnPlPackage = await import('@zxcvbn-ts/language-pl');
+		const zxcvbnPtBrPackage = await import('@zxcvbn-ts/language-pt-br');
+
+		const options: OptionsType = {
+			translations: zxcvbnEnPackage.translations,
+			graphs: zxcvbnCommonPackage.adjacencyGraphs,
+			dictionary: {
+				...zxcvbnCommonPackage.dictionary,
+				...zxcvbnCsPackage.dictionary,
+				...zxcvbnDePackage.dictionary,
+				...zxcvbnEsEsPackage.dictionary,
+				...zxcvbnFiPackage.dictionary,
+				...zxcvbnFrPackage.dictionary,
+				...zxcvbnIdPackage.dictionary,
+				...zxcvbnItPackage.dictionary,
+				...zxcvbnJaPackage.dictionary,
+				...zxcvbnNlBePackage.dictionary,
+				...zxcvbnPlPackage.dictionary,
+				...zxcvbnPtBrPackage.dictionary,
+				...zxcvbnEnPackage.dictionary,
+
+				// Check inputs
+				userInputs: [password?.value ?? '', email?.value ?? '', confirm_password?.value ?? '']
+			},
+			useLevenshteinDistance: true
+		};
+
+		zxcvbnOptions.setOptions(options);
+	};
+
+	$effect.pre(() => {
+		// register_languages();
+	});
+
+	// bind:value needs to be defined first to bind
+	// otherwise shows: typeerror $.get(...) is undefined
+	let email = $state<FormFieldState>({ value: "", error: new Array<string>() }),
+		password = $state<FormFieldState>({ value: "", error: new Array<string>() }),
+		confirm_password = $state<FormFieldState>({ value: "", error: new Array<string>() });
```

Comment:
Do you need to declare types of state here?

Reply:
Resolved on https://github.com/baseplate-admin/coreproject-v4-ui/pull/35/commits/369f9538507b82a0103db11af92d5af9ae69d724